### PR TITLE
Table: Matches column names with unescaped regex characters

### LINKED ISSUE #21106: Azure Log Analytics not formatting table cells if title includes ( ) characters
**What happened**:
Azure Log Analytics table cells not formatted if they contain ( ) characters. I have underlined the column name in red in the screen shots.

image

image

**What you expected to happen**:
Cell to be formatted based on the Visualization settings above.

**How to reproduce it (as minimally and precisely as possible)**:
See screen shots for example. Add a query to Azure Log Analytics which outputs column names including ( ) characters and set the visualization for the column to color the cells.

**Environment**:
- Grafana version: 6.5.1
- Data source type & version: Azure Application Insights
- OS Grafana is installed on: Ubuntu 18.04
- User OS & Browser: Windows 10 & Chrome

### PR DESCRIPTION
**Pull request by hugohaggmark**
**What this PR does / why we need it**:
This PR will make it so we can match column names with `unescaped regex` characters that are not used as a proper `regex` pattern.

**Column name**   | **Pattern that will match**
Requests (Failed)    | Requests (Failed)
Requests (Failed)    | Requests \\(Failed\\)
Requests (Failed)    | /Requests \\(Failed\\)/
Requests (Failed)    | /.*/

**Which issue(s) this PR fixes**:
Fixes #21106

**Special notes for your reviewer**:

### CODE REVIEW

File: packages/grafana-data/src/text/string.ts [hunk 1]
```diff
@@ -1,6 +1,32 @@
...
+export const unEscapeStringFromRegex = (value: string) => {
+  if (!value) {
+    return value;
+  }
+
+  return specialChars.reduce((escaped, currentChar) => escaped.replace('\\' + currentChar, currentChar), value);
+};
+
+export function stringStartsAsRegEx(str: string): boolean {
+  if (!str) {
```
Comment (dprokop) [inline packages/grafana-data/src/text/string.ts]: What about `str === ''`?

File: packages/grafana-data/src/text/string.ts [hunk 2]
```diff
@@ -1,6 +1,32 @@
+const specialChars = ['(', '[', '{', '}', ']', ')', '|', '*', '+', '-', '.', '?', '<', '>', '#', '&', '^', '$'];
```
Comment (dprokop) [inline packages/grafana-data/src/text/string.ts]: Great stuff with moving these utils to grafana-data

**Review by dprokop — APPROVED** [review]

File: packages/grafana-data/src/text/string.ts (hunk 1 above)
Comment (hugohaggmark) [reply packages/grafana-data/src/text/string.ts]: What about it :)

Comment (dprokop) [reply packages/grafana-data/src/text/string.ts]: I think this is more verbose than `!str` :) WDYT?

Comment (hugohaggmark) [reply packages/grafana-data/src/text/string.ts]: My thinking is that this takes care of the case where one tries to call this with an undefined string. Should add a test for this perhaps?

Comment (dprokop) [reply packages/grafana-data/src/text/string.ts]: Hmm, TS should take care of this with the `str: string` typing

Comment (hugohaggmark) [reply packages/grafana-data/src/text/string.ts]: Agree but this is string that comes from the UI, might not always be true that it is a string. Just being super cautious.

Comment (dprokop) [reply packages/grafana-data/src/text/string.ts]: Ok, that makes sense

**Pull request merged**